packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -363,6 +407,91 @@ contract OPContractsManager is ISemver {
         return output;
     }
 
+    function addGameType(AddGameInput[] memory _gameConfigs) external returns (AddGameOutput[] memory) {
+        if (address(this) == address(thisOPCM)) revert OnlyDelegatecall();
+
+        AddGameOutput[] memory outputs = new AddGameOutput[](_gameConfigs.length);
+        Blueprints memory bps = thisOPCM.blueprints();
+
+        for (uint256 i = 0; i < _gameConfigs.length; i++) {
+            AddGameInput memory gameConfig = _gameConfigs[i];
+
+            IPermissionedDisputeGame pdg = IPermissionedDisputeGame(
+                address(
+                    IDisputeGameFactory(gameConfig.systemConfig.disputeGameFactory()).gameImpls(
+                        GameTypes.PERMISSIONED_CANNON
+                    )
+                )
+            );
```

Comment:
gameImpls returns the type `IDisputeGame` type, so I don't think you need these casting wrappers.

Reply:
i need the permissioned dispute game to call .proposer() and .challenger(). i added a new local variable to case to IPDG in the permissioned dispute game case.

Reply:
I need a `FaultDisputeGame` to call `l2chainId`, which `IPermissionlessDisputeGame` is. I can clarify this to only cast to IPermissioned in the event gameConfig.permissioned is true, but I'll still need to case.